### DISCUSSION TITLE
How to detect typos?

### DISCUSSION THREAD

**Discussion by guettli** [a-h/templ#219] [Q&A] [answered]
Imagine I have a typo

```
package main

templ headerTemplate(name string) {
    <h1>{ name }<h1>
}
```

In above example the closing tag is not closing.

I am a bit paranoid and would like to have an automated way to detect typos like this.

Does templ help here, or do you know tools that could help?

**Comment by a-h** [comment, marked as answer]
The `templ generate` command will complain about the unclosed element.

In addition, templ has editor extensions for VS Code and Neovim that display those errors while you're working in your editor. (They're called "LSP diagnostics")

Support for Jet Brains IDEs is on its way.

**Reply by guettli** [reply]
sounds good. Thank you
